cirq-core/cirq/study/sweeps.py

```diff
@@ -276,6 +276,64 @@ def _from_json_dict_(cls, factors, **kwargs):
         return Product(*factors)
 
 
+class Concat(Sweep):
+    """Concatenates multiple sweeps along a new axis.
+
+    All sweeps must share the same descriptors.
+
+    If one sweep assigns 'a' to the values 0, 1, 2, and another sweep assigns
+    'a' to the values 3, 4, 5, the concatenation produces a sweep assigning
+    'a' to the values 0, 1, 2, 3, 4, 5 in sequence.
+    """
+
+    def __init__(self, *sweeps: Sweep) -> None:
+        if not sweeps:
+            raise ValueError("Concat requires at least one sweep.")
+
+        # Validate consistency across sweeps
+        first_sweep = sweeps[0]
+        for sweep in sweeps[1:]:
+            if sweep.keys != first_sweep.keys:
+                raise ValueError("All sweeps must have the same descriptors.")
+
+        self.sweeps = sweeps
+
+    def __eq__(self, other):
+        if not isinstance(other, Concat):
+            return NotImplemented
+        return self.sweeps == other.sweeps
+
+    def __hash__(self):
+        return hash(tuple(self.sweeps))
+
+    @property
+    def keys(self) -> List['cirq.TParamKey']:
+        return self.sweeps[0].keys if self.sweeps else []
```

Comment:
Nit - self.sweeps are not empty per check in the `__init__`
```suggestion
        return self.sweeps[0].keys
```

Reply:
good catch!